src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/CodeActions/Razor/ExtractToComponentCodeActionProvider.cs

```diff
@@ -78,16 +78,24 @@ private static (SyntaxNode? Start, SyntaxNode? End) GetStartAndEndElements(Razor
             return (null, null);
         }
 
-        var startElementNode = owner.FirstAncestorOrSelf<SyntaxNode>(IsBlockNode);
-
+        var startElementNode = GetBlockOrTextNode(owner);
         if (startElementNode is null || LocationInvalid(context.StartAbsoluteIndex, startElementNode))
         {
             return (null, null);
         }
 
-        var endElementNode = context.StartAbsoluteIndex == context.EndAbsoluteIndex
-            ? startElementNode
-            : GetEndElementNode(context, syntaxTree);
+        var hasSelection = context.StartAbsoluteIndex != context.EndAbsoluteIndex;
+
+        // In cases where the start element is just a text literal and there
+        // is no user selection avoid extracting the whole text literal.
+        if (startElementNode is MarkupTextLiteralSyntax && !hasSelection)
```

Comment:
Test for this?

Reply:
There's one already. This just keeps it passing